### ISSUE TITLE
Translate NEMO to Chinese

### ISSUE THREAD

**Issue by CairoLee** [MStr3am/NEMO#85]
Hello @MStr3am 

NEMO is an very nice project.
I want to help NEMO translate to chinese. Are you interested in?

**Comment by MStr3am** [comment]
Well all you need to do is to change the replace values in TranslateClient.txt file inside Patches folder.

**Comment by CairoLee** [comment]
I meat translate nemo's UI. No meat the ragexe message string. XD

**Comment by CairoLee** [comment]
Or i can just fork the project, and translate it.
but NEMO.exe is unfriendly to UTF8, could you try to support this?

![qq20151220-0](https://cloud.githubusercontent.com/assets/4458917/11916287/18a93326-a70b-11e5-8660-be62b358124e.png)

i guess nemo.exe is write by Qt. maybe can try this:

```
QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
```

**Comment by MStr3am** [comment]
If i use that codec I get issues for Non-ASCII characters=.

**Comment by CairoLee** [comment]
Oh... Has any idea to do this?

**Comment by CairoLee** [comment]
Can you to pick up all the string in the program, and put them in a configuration file? Like some lang_en-us.conf or lang_zh-CN.conf ?

**Comment by MStr3am** [comment]
hmm doesnt your screenshot already show Chinese why do you say its not supporting UTF

**Comment by dastgir** [comment]
@MStr3am see the header of those lists,
It's showing in ascii

**Comment by MStr3am** [comment]
do u have a translated _patchlist.qs. want to test out codec.

**Comment by CairoLee** [comment]
I try to translate _patchlist.qs into chinese, it's work find.
But i want to translate the listview header, and the other string...

![nemo](https://cloud.githubusercontent.com/assets/4458917/12053264/48c03448-af52-11e5-98ee-306a1de73d1f.png)

**Comment by Angelmelody** [comment]
CairoLee is trying to say he wanna translate the text strings on Neo UI into Chinese by hexing, 
but those text strings seems not support unicode encoding, and will become gibberish,So he 
request you to make it supporting unicode encoding text on NeO UI .

**Comment by MStr3am** [comment]
yea got it I am working on it

**Comment by CairoLee** [comment]
@Angelmelody 谢谢，喵玲儿。我就是这个意思。
@MStr3am Thank you. Sorry for my bad english. XD

**Comment by MStr3am** [comment]
Here is a screenshot with Google Translated strings. All the labels shown are read from a seperate INI file.
![snapshot_lang](https://cloud.githubusercontent.com/assets/4068357/12064125/2358ce1a-afe1-11e5-8850-a3f135d79809.PNG)

As for the buttons below those words are from Images so I am loading the image filenames through the INI to help with the translation.

**Comment by MStr3am** [comment]
-_- oops its translated to japanese instead of chinese anyways you get the idea.

**Comment by CairoLee** [comment]
Cool
 I will test it later

**Comment by MStr3am** [comment]
need to checkin first :P

**Comment by CairoLee** [comment]
it's work very fine! 
i hope neo can provide more customizable strings, like this!!~~~

![nemo01](https://cloud.githubusercontent.com/assets/4458917/12064686/d179c648-b004-11e5-9d89-843135d30186.png)

![nemo02](https://cloud.githubusercontent.com/assets/4458917/12064687/d89b2dc2-b004-11e5-9e78-2cb40944422a.png)

![nemo03](https://cloud.githubusercontent.com/assets/4458917/12064688/dceed522-b004-11e5-8f6c-435bca9b65fe.png)

**Comment by MStr3am** [comment]
ah tooltips ok

**Comment by CairoLee** [comment]
I'm looking forward to it. XD

**Comment by secretdataz** [comment]
suggestion : 
Suffix UI.ini with locale to support multiple language without having to edit the main file itself.
for example : UI_en_US.ini, UI_zh_CN.ini

**Comment by dastgir** [comment]
You mean prefix?

Also..
And load all ini? And have option which file to use?

**Comment by secretdataz** [comment]
Ah yes, what I mean is add option to select languages from the menu.
You don't have to load all inis as they could be loaded on demand (maybe from an open file dialog).
And, no, it's suffix.

**Comment by MStr3am** [comment]
![langchanger](https://cloud.githubusercontent.com/assets/4068357/12070145/f9130ac8-b088-11e5-8427-c4ebcb617c92.png)

like this?

**Comment by CairoLee** [comment]
Right!

**Comment by MStr3am** [comment]
Pushed the changes. 
You just need to make a file of the format UI_<langname>.ini and add all the strings in there. 
Refer UI_English.ini and Template_UI.ini for all the different fields/keys that need to be filled.

The Browse dialogs will still show whatever is your locale since it is part of Windows and not from QT (so no control over the button Texts in them)

**Comment by CairoLee** [comment]
Thank you again @MStr3am . PASS THE TEST !  >3< 

But still has an new problem, please check it.

- useing [chinese _patchlist.qs](https://raw.githubusercontent.com/CairoLee/NEMO-CN/master/Patches/_patchlist.qs) and "apply selected", open the "PatchList.log" will see Japanese and Chinese are displayed as question marks.

![nemo04](https://cloud.githubusercontent.com/assets/4458917/12071166/da387c06-b0d7-11e5-9d57-bbfcb256257e.png)

**Comment by MStr3am** [comment]
hmm this is weird I am seeing the chinese characters as normal ascii in the table with your _patchlist.qs o_o

**Comment by CairoLee** [comment]
@MStr3am This is a normal phenomenon, you need use GB2312 to make them display correctly.

1. use notepad++ 
2. menu -> Encoding -> Character Set -> Chinese -> GB2312

![notepad](https://cloud.githubusercontent.com/assets/4458917/12073661/dffdd450-b164-11e5-92a1-95259185ad70.png)

**Comment by CairoLee** [comment]
If the Character Set is right, then you will see the correctly character.
Or you can just using google translate to get some chinese or Japanese character.

![notepad1](https://cloud.githubusercontent.com/assets/4458917/12073664/282031c4-b165-11e5-9fe0-4681780f72a4.png)

**Comment by MStr3am** [comment]
i meant the table in NEMO lol

**Comment by CairoLee** [comment]
yes, the listview table can showing the chinese characters as normal ascii, but after completing diff, can't see the correctly chinese characters in "PatchList.log".

**Comment by CairoLee** [comment]
@MStr3am Here is my PatchList.log file: 
[PatchList.zip](https://github.com/MStr3am/NEMO/files/78020/PatchList.zip)

**Comment by MStr3am** [comment]
hmm my patchlist log is coming fine though despite NEMO not showing the right encoding.

**Comment by CairoLee** [comment]
@MStr3am Sir,could you test it in chinese system?

Windows 7 Ultimate with Service Pack 1 (x64) - DVD (Chinese-Simplified)
SHA1 2CE0B2DB34D76ED3F697CE148CB7594432405E23
ed2k://|file|cn_windows_7_ultimate_with_sp1_x64_dvd_u_677408.iso|3420557312|B58548681854236C7939003B583A8078|/

**Comment by MStr3am** [comment]
would be better if we TeamView-ed . I don't want to install another Win7 just for testing this.

**Comment by CairoLee** [comment]
@MStr3am OK, How do we exchange the Teamviewer ID? Email or skype?

**Comment by CairoLee** [comment]
I'm ready to accept your control. Waiting for you response. XD

**Comment by CairoLee** [comment]
Please contact me, I will send you the Teamviewer ID, my email address is: cairoliyu#gmail.com (# to @)

**Comment by CairoLee** [comment]
Any news?

**Comment by 4144** [comment]
@CairoLee if you still need Nemo translations, see my fork at https://gitlab.com/4144/Nemo

Now nemo support utf-8 in all places, and allow translate almost all messages
See untranslated templates in https://gitlab.com/4144/Nemo/blob/master/Patches_Chinese%20Simplified.ini and https://gitlab.com/4144/Nemo/blob/master/UI_Chinese%20Simplified.ini

**Comment by CairoLee** [comment]
> @CairoLee if you still need Nemo translations, see my fork at https://gitlab.com/4144/Nemo
> 
> Now nemo support utf-8 in all places, and allow translate almost all messages
> See untranslated templates in https://gitlab.com/4144/Nemo/blob/master/Patches_Chinese%20Simplified.ini and https://gitlab.com/4144/Nemo/blob/master/UI_Chinese%20Simplified.ini

Great, i will follow this, thanks @4144
